### PR TITLE
Main fix

### PR DESCRIPTION
**Pull request by icode247**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <a href="https://www.typescriptlang.org/"><img alt="Typescript" src="https://img.shields.io/badge/TypeScript-007ACC?style=for-the-badge&logo=typescript&logoColor=white"/></a>
   <a href="https://vercel.com/"><img alt="Vercel" src="https://img.shields.io/badge/vercel-F22F46?style=for-the-badge&logo=vercel&logoColor=white"/></a>
   <a href="https://nextjs.org/"><img alt="Next.js" src="https://img.shields.io/badge/next.js-000000?style=for-the-badge&logo=nextdotjs&logoColor=white"/></a>
-  <a href="https://opensource.org/licenses/Apache-2.0"><img alt="LICENSE" src="https://img.shields.io/github/license/tigrisdata/tigris-vercel-starter.svg?style=for-the-badge"/></a>
+  <a href="https://opensource.org/licenses/MIT"><img alt="LICENSE" src="https://img.shields.io/badge/licence-MIT-orange?style=for-the-badge"></a></a>
```
Comment (leggetter) [inline README.md:8]: Duplicate `</a>`

File: pages/index.tsx [hunk 2]
```diff
@@ -1,150 +1,156 @@
...
-            <main>
-                <header>
-                    <nav className="navbar bg-body-tertiary">
-                        <div className="container-fluid">
-                            <a className="navbar-brand">Shop <i className="fa fa-shopping-bag" aria-hidden="true"></i>
-                            </a>
-                            <form className="d-flex" role="search">
-                                <input className="form-control me-2" type="search" placeholder="Search" aria-label="Search" />
-                            </form>
+  const [products, setProducts] = useState<Product[]>([]);
```
Comment (leggetter) [inline pages/index.tsx:14]: nit: should we be using async await over the `.then` promise syntax?

File: pages/index.tsx [hunk 3]
```diff
@@ -1,150 +1,156 @@
...
+              </nav>
+              <div className="row">
+                {products.length ? (
+                  products.map((product) => {
+                    const stars = Array.from(
+                      { length: product.star },
+                      (_, i) => <span key={i} className="fas fa-star"></span>
+                    );
+                    return (
+                      // eslint-disable-next-line react/jsx-key
```
Comment (leggetter) [inline pages/index.tsx:107]: Since a `key` has been added we can remove the eslint comment

File: README.md (hunk 1 above)
Comment (leggetter) [inline README.md:8]:
```suggestion
  <a href="https://opensource.org/licenses/MIT"><img alt="LICENSE" src="https://img.shields.io/badge/licence-MIT-orange?style=for-the-badge"></a>
```

**Review by leggetter — CHANGES_REQUESTED** [review]
Just one requested change and a few comments.

File: pages/index.tsx (hunk 3 above)
Comment (leggetter) [reply pages/index.tsx:107]: Same for the starter branch.

File: pages/index.tsx (hunk 2 above)
Comment (icode247) [reply pages/index.tsx:14]: Where?

Comment (icode247) [reply pages/index.tsx:14]: I guess you mean:

```
const fetchProducts = () => {
    fetch("/api/store")
      .then((response) => response.json())
      .then((data) => {
        if (data.result) {
          setProducts(data.result);
        }
      })
      .catch((e) => {
        console.log(`Error: ${e}`);
      });
  };

  const searchQuery = () => {
    fetch(`/api/store/search?query=${encodeURI(searchInput)}`, {
      method: "GET",
    })
      .then((response) => response.json())
      .then((data) => {
        if (data.result) {
          setProducts(data.result);
        }
      });
  };
```

Comment (icode247) [reply pages/index.tsx:14]: Since it's not deprecated, we should use it.

Comment (leggetter) [reply pages/index.tsx:14]: > I guess you mean:
> 
> ```
> const fetchProducts = () => {
>     fetch("/api/store")
>       .then((response) => response.json())
>       .then((data) => {
>         if (data.result) {
>           setProducts(data.result);
>         }
>       })
>       .catch((e) => {
>         console.log(`Error: ${e}`);
>       });
>   };
> 
>   const searchQuery = () => {
>     fetch(`/api/store/search?query=${encodeURI(searchInput)}`, {
>       method: "GET",
>     })
>       .then((response) => response.json())
>       .then((data) => {
>         if (data.result) {
>           setProducts(data.result);
>         }
>       });
>   };
> ```

Yes, I'm asking if we should update this to use `async` and `await` for easier to read code?

Comment (icode247) [reply pages/index.tsx:14]: Let's do that.

**Pull request closed without merging**